src/guide/using-the-cli.md

```diff
@@ -0,0 +1,151 @@
+
+<p class="tip"><strong>Important:</strong> Only confirm the handshake if the shown hash matches the one in your terminal.</p>
+
+#### Logging out
+
+Logging out from MachineLabs with the CLI is as simple as running the `login` command without further action:
+
+```
+$ ml logout
+```
+
+## Initializing Labs
+
+Once logged in, we can start interacting with the CLI. There are several scenarios where initializing a lab might be want we want. We either start a new project and have to initialize it eventually, so we can push it to the MachineLabs platform, or we already have an existing project we'd like to run on MachineLabs. For the latter, the CLI comes in very handy.
+
+To initialize a new or existing project, all we have to do is navigating into the projects folder and use the `init` command:
+
+```
+$ cd path/to/project
+$ ml init
+```
+
+This will create an `ml.yaml` file for our project, which is required to run a lab on MachineLabs. If our project already comes with an `ml.yaml` configuration file and we wish to override it, we can use the `--force` option:
+
+```
+$ ml init --force
+```
+
+If you want to learn more about what can be configured in a lab's `ml.yaml` file, head over to our guide on [Configuring Labs](configuring-labs.html).
+
+Alright, now that our project is initialized, we can push it up on MachineLabs to execute it there!
+
+## Saving and updating Labs
+
+Whether we're creating a new lab or taking an existing project, eventually, we want to push whatever we have and run it on the MachineLabs platform. That's where the CLI's `push` command comes into play. It lets either update a lab we own, or push an entirely new one, which will then be available on the web.
+
+#### Saving new Labs
+
+To create a new lab, we run the `push` command from within our project folder. We should also give it a `name` and a `description` using the dedicated options, so people on MachineLabs can easier understand what our lab is about:
+
+```
+$ ml push --name "My first lab" --description "This is my first lab, created using the CLI"
+```
+
+Note that as of right now, every lab needs a `main.py` file so it can be pushed to MachineLabs.
+
+Once pushed, the CLI will give us a link which points to the lab on MachineLabs - cool right?
+
+#### Updating Labs
+
+Often we want to update our labs on MachineLabs with the changes we've done on our local machine. The `push` command makes this possible as well. All we need to do is passing the `id` of the lab we want to push our changes to. So assuming we've changed the contents of our project's `main.py` and our lab's id is `4711`, we can now run the following command to push our current state to MachineLabs:
+
+```
+$ ml push --id 4711
```

Comment:
This wouldn't be needed anymore with the changes to the CLI that are currently under way. A `push` that created an id would put that id into the `ml.yaml` file and a `pull` would also add either the original id or generated id into the `ml.yaml` file depending on whether you are the owner of that lab or not.

So, the id becomes more or less transparent for the user.

Reply:
That's true. I still think it'd be good to have this documented. I'll add a section that clarifies that `id` is actually optional when `ml.yaml` comes with a dedicated `cli` section. This stuff I haven't touched on yet at all.